src/message.ts

```diff
@@ -119,32 +119,39 @@ export async function verifyMessageSignature(message: Message, didResolver: DIDR
 
   // use a set to do O(1) "lookups" for the correct id
   const verificationMethodIds = new Set();
-  let publicKey;
+  let verificationMethod: VerificationMethod;
 
-  for (let verificationMethod of verificationMethods) {
-    verificationMethodIds.add(verificationMethod.id);
+  for (const vm of verificationMethods) {
+    verificationMethodIds.add(vm.id);
 
-    // TODO: figure out if we need to support ALL verification method properties
-    //       listed here: https://www.w3.org/TR/did-spec-registries/#verification-method-properties
-
-    // TODO: figure out if we need to support ALL verification method types
-    //       listed here: https://www.w3.org/TR/did-spec-registries/#verification-method-types
-
-    // TODO: figure out if we need to check to ensure that `controller` === did in kid
-    //       are the same. This may matter more for a `PermissionsRequest`
-
-    // going to naively yank `publicKeyJwk` for now until above TODOS are resolved. thxbai
     if (verificationMethodIds.has(kid)) {
-      publicKey = verificationMethod.publicKeyJwk;
+      verificationMethod = vm;
       break;
     }
   }
 
-  if (!publicKey) {
-    throw new Error('failed to find respective public key to verify signature');
+  if (!verificationMethod) {
+    throw new Error('public key needed to verify signature not found in DID Document');
+  }
+
+  // more info about the `JsonWebKey2020` type can be found here:
+  // https://www.w3.org/TR/did-spec-registries/#jsonwebkey2020
+  if (verificationMethod.type !== 'JsonWebKey2020') {
+    throw new Error(`verification method [${kid}] must be JsonWebKey2020`);
   }
 
-  const result = await jws.verify(attestation, publicKey);
+  const { publicKeyJwk } = verificationMethod;
+
+  // more info about the `publicKeyJwk` property can be found here:
+  // https://www.w3.org/TR/did-spec-registries/#publickeyjwk
+  if (!publicKeyJwk) {
+    throw new Error(`publicKeyJwk property not found on verification method [${kid}]`);
+  }
```

Comment:
Need clarification: why these validations are done in place here, but not other validations? Based on our previous discussions, I thought you are in favor of JSON schema based validation, if so why the exception here?

Reply:
Just as a placeholder. I don't want to forget that we decided to only support `JsonWebKey2020` and `publicKeyJwk`. I went ahead and added tests for this as well so that we remember to consider this when we tackle validation

Reply:
I feel this warrants a TODO like you had with an issue number attached, because right now it doesn't look like temporary code.

e.g.
// TODO: #123 - Need to replace these validation logic with schema based validation

Reply:
added TODOs in the code and created a GH to create issues for each TODO: #32